client/client_test.go

```diff
@@ -250,3 +250,54 @@ func TestDoRemovesDoubleSlashesFromFilesAPI(t *testing.T) {
 	err = c.Do(context.Background(), "GET", "/api/2.0/anotherservice//test", nil, map[string]any{}, nil)
 	require.NoError(t, err)
 }
+
+func TestNonJSONResponseIncludedInError(t *testing.T) {
+	type testCase struct {
+		statusCode   int
+		status       string
+		errorMessage string
+	}
+	cases := []testCase{
+		{
+			statusCode:   400,
+			status:       "Bad Request",
+			errorMessage: "unexpected response from server (Bad Request): invalid character '<' looking for beginning of value (original: <html><body>hello</body></html>)",
```

Comment:
When we're changing this anyway, this is a good opportunity to remove the boilerplate if this is indeed an HTML body. We can check if the error is equal to `json.SyntaxError`, and if so, just dump the HTML and not the string `invalid character...`.